Jenkinsfile

```diff
@@ -232,14 +232,36 @@ try {
           }
         }
 
-        stage("memcheckReleaseTest") {
-          // Runs separately as it depends on clang-release to be built
-          if (env.BRANCH_NAME == 'master' || full_ci) {
-            sh "mkdir ./clang-release-memcheck-test"
-            // If this shows a leak, try --leak-check=full, which is slower but more precise
-            sh "valgrind --tool=memcheck --error-exitcode=1 --gen-suppressions=all --num-callers=25 --suppressions=resources/.valgrind-ignore.txt ./clang-release/hyriseTest clang-release-memcheck-test --gtest_filter=-NUMAMemoryResourceTest.BasicAllocate"
-          } else {
-            Utils.markStageSkippedForConditional("memcheckReleaseTest")
+        parallel memcheckReleaseTest: {
+          stage("memcheckReleaseTest") {
+            // Runs separately as it depends on clang-release to be built
+            if (env.BRANCH_NAME == 'master' || full_ci) {
+              sh "mkdir ./clang-release-memcheck-test"
+              // If this shows a leak, try --leak-check=full, which is slower but more precise
+              sh "valgrind --tool=memcheck --error-exitcode=1 --gen-suppressions=all --num-callers=25 --suppressions=resources/.valgrind-ignore.txt ./clang-release/hyriseTest clang-release-memcheck-test --gtest_filter=-NUMAMemoryResourceTest.BasicAllocate"
+            } else {
+              Utils.markStageSkippedForConditional("memcheckReleaseTest")
+            }
+          }
+        }, tpchQueryPlans: {
+          stage("tpchQueryPlans") {
+            // Query plan generation runs as part of this parallel block in order to avoid a load imbalance between the parallel blocks
```

Comment:
I don't really understand this.

Reply:
I thought I'd explain why it is not part of that big `parallel` block but runs towards the end. In the end, it probably doesn't even matter. Removed the comment so that it doesn't confuse anyone.